sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala

```diff
@@ -184,4 +185,38 @@ case class GpuGetMapValue(child: Expression, key: Expression)
   override def left: Expression = child
 
   override def right: Expression = key
+}
+
+case class GpuArrayContains(left: Expression, right: Expression)
+  extends GpuBinaryExpression with ImplicitCastInputTypes with NullIntolerant {
+
+  override def dataType: DataType = BooleanType
+
+  override def inputTypes: Seq[AbstractDataType] = {
+    (left.dataType, right.dataType) match {
+      case (ArrayType(e1, hasNull), e2) =>
+        findTightestCommonType(e1, e2) match {
+          case Some(dt) => Seq(ArrayType(dt, hasNull), dt)
+        }
+      case _ => Seq.empty
+    }
+  }
+
+  override def nullable: Boolean = {
+    left.nullable || right.nullable || left.dataType.asInstanceOf[ArrayType].containsNull
+  }
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: Scalar): ColumnVector =
+    lhs.getBase.listContains(rhs)
+
+  override def doColumnar(numRows: Int, lhs: Scalar, rhs: Scalar): ColumnVector =
+    throw new IllegalStateException("This is not supported yet")
```

Comment:
The we need to be explicit in the type checking that we do not support literal arrays.